### PR TITLE
Sync reference docs and CL for preview.7 release

### PR DESCRIPTION
**Pull request by michaeltlombardi**
# PR Summary

This change syncs the changes and new reference documentation for the `preview.7` release of DSC. It also updates the changelog for the release.

This change:

- Fixes [AB#243046](https://dev.azure.com/msft-skilling/cebd7ef5-4282-448b-9701-88c8637581b7/_workitems/edit/243046)
- Fixes [AB#250844](https://dev.azure.com/msft-skilling/cebd7ef5-4282-448b-9701-88c8637581b7/_workitems/edit/250844)
- Fixes [AB#250845](https://dev.azure.com/msft-skilling/cebd7ef5-4282-448b-9701-88c8637581b7/_workitems/edit/250845)

## PR Checklist

- [x] **Descriptive Title:** This PR's title is a synopsis of the changes it proposes.
- [x] **Summary:** This PR's summary describes the scope and intent of the change.
- [x] **Contributor's Guide:** I have read the [contributors guide][contrib].
- [x] **Style:** This PR adheres to the [style guide][style].

[contrib]: https://learn.microsoft.com/powershell/scripting/community/contributing/overview
[style]: https://learn.microsoft.com/powershell/scripting/community/contributing/powershell-style-guide

### CODE REVIEW

File: dsc/docs-conceptual/dsc-3.0/changelog.md [hunk 1]
```diff
@@ -22,10 +45,514 @@ All notable changes to DSCv3 are documented in this file. The format is based on
...
+[p7-09]: reference/schemas/outputs/resource/list.md#capability-delete
+[p7-10]: reference/schemas/resource/manifest/root.md#schema-1
+[p7-11]: reference/schemas/config/metadata.md
+[p7-12]: reference/schemas/resource/manifest/delete.md#json-input-argument
+[p7-13]: reference/schemas/resource/manifest/export.md#json-input-argument
+[p7-14]: reference/schemas/resource/manifest/get.md#json-input-argument
+[p7-15]: reference/schemas/resource/manifest/set.md#json-input-argument
+[p7-16]: reference/schemas/resource/manifest/test.md#json-input-argument
+[p7-17]: reference/schemas/resource/manifest/validate.md#json-input-argument
+[p7-18]: reference/schemas/config/resource.md#dependsOn
```
Comment (sdwheeler) [inline dsc/docs-conceptual/dsc-3.0/changelog.md:540]:
```suggestion
[p7-18]: reference/schemas/config/resource.md#dependson
```

File: dsc/docs-conceptual/dsc-3.0/changelog.md [hunk 2]
```diff
@@ -594,6 +1162,19 @@ in this release, see the [diff on GitHub][compare-v3.0.0-alpha.2].
 
   </details>
 
+<!-- alpha.2 links -->
+[a2.01]: reference/schemas/config/resource.md#dependson
+[a2.02]: reference/schemas/resource/manifest/export.md
+[a2.05]: reference/cli/resource/get.md##a---all
```
Comment (sdwheeler) [inline dsc/docs-conceptual/dsc-3.0/changelog.md:1168]:
```suggestion
[a2.05]: reference/cli/resource/get.md#-a---all
```

File: dsc/docs-conceptual/dsc-3.0/reference/schemas/outputs/resource/list.md [hunk 3]
```diff
@@ -65,6 +77,62 @@ Required: true
...
+
+The following list describes the available capabilities for a resource:
+
+- <a id="capability-get" /> `Get` - The resource supports retrieving the current state of an
+  instance. All DSC Resources must have this capability. A resource has this capability when it
+  defines the mandatory [get][07] property in its resource manifest.
+- <a id="capability-set" /> `Set` - The resource supports enforcing the desired state of an
+  instance. A resource has this capability when it defines the [set][08] property in its resource
+  manifest. Resources without this capability can't be used with the [dsc resource set][09] or
+  [dsc config set][10] commands unless they're in a [Microsoft.DSC/Assertion][11] group as a nested
```
Comment (sdwheeler) [inline dsc/docs-conceptual/dsc-3.0/reference/schemas/outputs/resource/list.md:94]: Can't figure out what `[Microsoft.DSC/Assertion][11]` is supposed to point to. The current definition is 404.

File: dsc/docs-conceptual/dsc-3.0/reference/schemas/resource/manifest/root.md [hunk 4]
```diff
@@ -249,22 +291,23 @@ resource. This property must always be an object that defines one of the followi
...
+[03]: ../../../cli/config/export.md
+[04]: ../../../cli/resource/export.md
+[05]: ../../../cli/resource/get.md
+[06]: ../../../cli/resource/get.md#-a---all
+[07]: export.md
+[08]: get.md
+[09]: set.md
+[10]: test.md
+[11]: validate.md
+[12]: provider.md
```
Comment (sdwheeler) [inline dsc/docs-conceptual/dsc-3.0/reference/schemas/resource/manifest/root.md:312]: You link to `provider.md` but this PR deletes that file.

**Review by sdwheeler — CHANGES_REQUESTED** [review]
See my comments

File: dsc/docs-conceptual/dsc-3.0/reference/schemas/resource/manifest/root.md (hunk 4 above)
Comment (michaeltlombardi) [reply dsc/docs-conceptual/dsc-3.0/reference/schemas/resource/manifest/root.md:312]:
```suggestion
[12]: adapter.md
```

File: dsc/docs-conceptual/dsc-3.0/reference/schemas/outputs/resource/list.md (hunk 3 above)
Comment (michaeltlombardi) [reply dsc/docs-conceptual/dsc-3.0/reference/schemas/outputs/resource/list.md:94]:
```suggestion
  [dsc config set][10] commands unless they're in a `Microsoft.DSC/Assertion` group as a nested
```

**Pull request merged**